build-go.sh

```diff
@@ -0,0 +1,231 @@
+#!/bin/bash
```

Comment:
this shell file is intense. did you lift it or make it from scratch? 

cc @chriscool for some quick review?

(no need for `sh`, I'm ok to keep `bash` to save time here. all of the people releasing go-ipfs have `bash` installed.)